pkg/common/rotationutil/rotationutil_test.go

```diff
@@ -12,31 +15,88 @@ import (
-	goodCert, _, err := util.SelfSign(temp)
-	require.NoError(t, err)
-
-	// Cert is brand new
-	assert.False(t, ShouldRotateX509(mockClk.Now(), goodCert))
-
-	// Cert that's almost expired
-	temp.NotBefore = mockClk.Now().Add(-1 * time.Hour)
-	temp.NotAfter = mockClk.Now().Add(1 * time.Minute)
-	badCert, _, err := util.SelfSign(temp)
-	require.NoError(t, err)
-
-	assert.True(t, ShouldRotateX509(mockClk.Now(), badCert))
-
-	// Cert that's already expired
-	temp.NotBefore = mockClk.Now().Add(-1 * time.Hour)
-	temp.NotAfter = mockClk.Now().Add(-1 * time.Minute)
-	badCert, _, err = util.SelfSign(temp)
-	require.NoError(t, err)
-
-	assert.True(t, ShouldRotateX509(mockClk.Now(), badCert))
+	log, _ := testlog.NewNullLogger()
+
+	for _, tc := range []struct {
+		desc               string
+		makeTemplate       func() (*x509.Certificate, error)
+		availabilityTarget time.Duration
+		shouldRotate       bool
+	}{
+		{
+			desc: "brand new cert",
+			makeTemplate: func() (*x509.Certificate, error) {
+				return util.NewSVIDTemplate(mockClk, "spiffe://example.org/test")
+			},
+			shouldRotate: false,
+		},
+		{
+			desc: "cert that's almost expired",
+			makeTemplate: func() (*x509.Certificate, error) {
+				temp, err := util.NewSVIDTemplate(mockClk, "spiffe://example.org/test")
+				if err != nil {
+					return nil, err
+				}
+				temp.NotBefore = mockClk.Now().Add(-1 * time.Hour)
+				temp.NotAfter = mockClk.Now().Add(1 * time.Minute)
+				return temp, nil
+			},
+			shouldRotate: true,
+		},
+		{
+			desc: "cert that's already expired",
+			makeTemplate: func() (*x509.Certificate, error) {
+				temp, err := util.NewSVIDTemplate(mockClk, "spiffe://example.org/test")
+				if err != nil {
+					return nil, err
+				}
+				temp.NotBefore = mockClk.Now().Add(-1 * time.Hour)
+				temp.NotAfter = mockClk.Now().Add(-11 * time.Minute)
+				return temp, nil
+			},
+			shouldRotate: true,
+		},
+		{
+			desc: "rotation by availability_target",
+			makeTemplate: func() (*x509.Certificate, error) {
+				temp, err := util.NewSVIDTemplate(mockClk, "spiffe://example.org/test")
+				if err != nil {
+					return nil, err
+				}
+				temp.NotBefore = mockClk.Now().Add(-24 * time.Hour)
+				temp.NotAfter = mockClk.Now().Add(48 * time.Hour)
+				return temp, nil
+			},
+			availabilityTarget: 48 * time.Hour,
+			shouldRotate:       true,
+		},
+		{
+			desc: "x509_svid_ttl isn't long enough to guarantee the availability_target",
+			makeTemplate: func() (*x509.Certificate, error) {
+				temp, err := util.NewSVIDTemplate(mockClk, "spiffe://example.org/test")
+				if err != nil {
+					return nil, err
+				}
+				temp.NotBefore = mockClk.Now().Add(-6 * time.Hour)
+				temp.NotAfter = mockClk.Now().Add(24 * time.Hour)
+				return temp, nil
+			},
+			availabilityTarget: 24 * time.Hour,
+			shouldRotate:       false,
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			temp, err := tc.makeTemplate()
+			require.NoError(t, err)
+
+			cert, _, err := util.SelfSign(temp)
```

Comment:
NIT: signing a x.509 certificate looks like an overkill.. we only need `cert.NotBefore` & `cert.NotAfter`...
maybe you can create a x509.Certificate struct with those values set?